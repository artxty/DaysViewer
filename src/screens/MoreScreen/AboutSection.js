import React from 'react';
import { Text } from 'react-native';

import ListItem from '../../components/MoreListItem';

export default function getAboutSection() {
  return {
    title: 'About',
    data: [
      <ListItem label="Rate app">
        <Text>text</Text>
      </ListItem>,
      <ListItem label="Share with friends">
        <Text>text</Text>
      </ListItem>,
      <ListItem label="Privacy">
        <Text>text</Text>
      </ListItem>,
    ],
  };
}
