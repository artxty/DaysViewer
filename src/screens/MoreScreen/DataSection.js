import React from 'react';
import { Button } from 'react-native';
import * as FileSystem from 'expo-file-system';
import * as Sharing from 'expo-sharing';
import * as DocumentPicker from 'expo-document-picker';

import ListItem from '../../components/MoreListItem';
import daysStorage from '../../storage/daysStorage';
import appInfoStorage from '../../storage/appInfoStorage';

export default function getDataSection() {
  return {
    title: 'Data',
    data: [
      <ListItem label="Export">
        <Button
          title="Share backup file"
          onPress={async () => {
            const keys = await daysStorage.getAllKeys();
            const data = await daysStorage.get(keys);
            const uri = `${FileSystem.cacheDirectory}backup.json`;
            await FileSystem.writeAsStringAsync(uri, JSON.stringify(data));
            await Sharing.shareAsync(uri);
          }}
        />
      </ListItem>,
      <ListItem label="Import">
        <Button
          title="Choose backup file"
          onPress={async () => {
            const document = await DocumentPicker.getDocumentAsync({ type: 'application/json' });
            if (document.type === 'success') {
              const jsonString = await FileSystem.readAsStringAsync(document.uri);
              /* TODO: validate json */
              await daysStorage.clear();
              await daysStorage.set(JSON.parse(jsonString));
              await appInfoStorage.set('doesCalendarNeedUpdate', true);
            }
          }}
        />
      </ListItem>,
    ],
  };
}
