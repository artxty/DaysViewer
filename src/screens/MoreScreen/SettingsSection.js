import React from 'react';

import NotificationsListItem from '../../components/NotificationsListItem';
import PastMonthsRangeListItem from '../../components/PastMothsRangeListItem';

export default function getSettingsSection(useDailyNotification, usePastMonthsRange) {
  return {
    title: 'Settings',
    data: [
      /* Notifications */
      <NotificationsListItem hook={useDailyNotification} />,
      /* Past months range */
      <PastMonthsRangeListItem hook={usePastMonthsRange} />,
      /* Next one */
      // <ListItem label="DarkMode">
      //   <Text>text</Text>
      // </ListItem>,
    ],
  };
}
