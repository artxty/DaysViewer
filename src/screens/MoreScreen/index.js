import React, { useContext } from 'react';

import MoreList from '../../components/MoreList';
import getSettingsSection from './SettingsSection';
import getDataSection from './DataSection';
// import getAboutSection from './AboutSection';
import { SettingsContext } from '../../storage/settingsContext';

export default function MoreScreen() {
  const { useDailyNotification, usePastMonthsRange } = useContext(SettingsContext);

  const settingsSection = getSettingsSection(
    useDailyNotification,
    usePastMonthsRange,
  );

  return (
    <MoreList
      sections={[
        settingsSection,
        getDataSection(),
        // getAboutSection(),
      ]}
    />
  );
}

MoreScreen.navigationOptions = {
  header: null,
};
