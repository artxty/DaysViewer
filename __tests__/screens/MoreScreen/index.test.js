import React from 'react';
import renderer from 'react-test-renderer';

import MoreScreen from '../../../src/screens/MoreScreen';

jest.mock('../../../src/components/MoreList', () => 'MoreList');
jest.mock('react', () => ({
  ...jest.requireActual('react'),
  useContext: () => ({
    useDailyNotification: [{}, jest.fn()],
    usePastMonthsRange: [3, jest.fn()],
  }),
}));

describe('MoreScreen', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<MoreScreen />);
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
