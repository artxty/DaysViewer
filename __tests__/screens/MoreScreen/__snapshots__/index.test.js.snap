// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoreScreen renders correctly 1`] = `
<MoreList
  sections={
    Array [
      Object {
        "data": Array [
          <NotificationsListItem
            hook={
              Array [
                Object {},
                [MockFunction],
              ]
            }
          />,
          <pastMonthsRangeListItem
            hook={
              Array [
                3,
                [MockFunction],
              ]
            }
          />,
        ],
        "title": "Settings",
      },
      Object {
        "data": Array [
          <ListItem
            label="Export"
            onPress={[Function]}
            style={Object {}}
          >
            <Button
              onPress={[Function]}
              title="Share backup file"
            />
          </ListItem>,
          <ListItem
            label="Import"
            onPress={[Function]}
            style={Object {}}
          >
            <Button
              onPress={[Function]}
              title="Choose backup file"
            />
          </ListItem>,
        ],
        "title": "Data",
      },
    ]
  }
/>
`;
